City-of-Helsinki/berth-reservations-admin
# VEN-898 | Persist list view tables' sorting and filtering states

## Description :sparkles:

Sorting and filtering options are persisted for each list view even after navigating to and from their respective pages.

Additionally implements a new type of filter for application pages, which enables filtering by application status.

### Closes :no_good_woman:

[VEN-898](https://helsinkisolutionoffice.atlassian.net/browse/VEN-898)

## Testing :alembic:

### Automated tests :gear:️

- `browser-tests/listViewTableStatePersistence.ts`
- Updated relevant unit tests

### Manual testing :construction_worker_man:

Persistent table states:

1. Go to any of the listing pages:
    - ApplicationList
    - CustomerList
    - HarborList
    - UnmarkedWsNoticeList
    - WinterStorageApplicationList
    - WinterStorageAreaList
2. Change column sorting and or any additional filters the page might have
3. Navigate away from the page
4. When returning to the page, any settings changed in step 4. should still be present

Filtering by application status:

1. Go to any application page:
    - ApplicationList
    - UnmarkedWsNoticeList
    - WinterStorageApplicationList
2. Change the filter in the top right corner of the table
4. Only applications in the selected state should be displayed in the table
5. Correct count of items in the table should be displayed in the top left corner
6. These changes should also persist when navigating to and from the page
